animation_nodes/nodes/color/c_utils.pyx

```diff
@@ -0,0 +1,66 @@
+import cython
+from ... math cimport (
+    addColor,
+    addColor_Inplace,
+    scaleColor_Inplace
+)
+from ... data_structures cimport (
+    Color,
+    LongList,
+    ColorList,
+    VirtualColorList,
+    PolygonIndicesList
+)
+
+def getLoopColorsFromVertexColors(PolygonIndicesList polygons, VirtualColorList colors):
+    cdef loopsCount = len(polygons.indices)
+    cdef ColorList loopsColors = ColorList(length = loopsCount)
+
+    cdef long i
+    for i in range(loopsCount):
+        loopsColors.data[i] = colors.get(polygons.indices[i])[0]
+    return loopsColors
+    
+def getLoopColorsFromPolygonColors(PolygonIndicesList polygons, VirtualColorList colors):
+    cdef ColorList loopsColors = ColorList(length = len(polygons.indices))
+    
+    cdef long i, j, index
+    cdef Color polygonColor
+    index = 0 
+    for i in range(len(polygons)):
+        polygonColor = colors.get(i)[0]
+        for j in range(polygons.polyLengths[i]):
+            loopsColors.data[index] = polygonColor
+            index += 1
+    return loopsColors
+
+@cython.cdivision(True)
+def getVertexColorsFromLoopColors(long vertexCount, PolygonIndicesList polygons, VirtualColorList colors):
+    cdef ColorList vertexsColors = ColorList.fromValue((0.0, 0.0, 0.0, 0.0), length = vertexCount)
```

Comment:
Instead of using `fromValue` to zero initialize the list, you can create the list, then use the fill function, like so:
```
    cdef ColorList vertexColors = ColorList(length = vertexCount)
    cdef LongList loopCounts = LongList(length = vertexCount)        
    vertexColors.fill(0)
    loopCounts.fill(0)
```

Also, `vertexsColors` should be `vertexColors`, there is no `s`.

Reply:
I have changed that.